visualization/app/codeCharta/state/effects/updateVisibleTopLabels/updateVisibleTopLabels.effect.ts

```diff
@@ -0,0 +1,22 @@
+import { Inject, Injectable } from "@angular/core"
+import { Store } from "../../angular-redux/store"
+import { createEffect } from "../../angular-redux/effects/createEffect"
+import { map, withLatestFrom } from "rxjs"
+import { visibleFileStatesSelector } from "../../selectors/visibleFileStates.selector"
+import { codeMapNodesSelector } from "../../selectors/accumulatedData/codeMapNodesSelector"
+import { setAmountOfTopLabels } from "../../store/appSettings/amountOfTopLabels/amountOfTopLabels.actions"
+import { getNumberOfTopLabels } from "./getNumberOfTopLabels"
+
+@Injectable()
+export class UpdateVisibleTopLabelsEffect {
+	updateVisibleTopLabels$ = createEffect(() =>
+		this.store.select(visibleFileStatesSelector).pipe(
+			withLatestFrom(this.store.select(codeMapNodesSelector)),
+			map(([, codeMapNodes]) => {
+				return setAmountOfTopLabels(getNumberOfTopLabels(codeMapNodes))
+			})
+		)
+	)
+
+	constructor(@Inject(Store) private store: Store) {}
```

Comment:
[Optional] I like the constructor before the effects and the effects I have written so far have it the way I like ;)